### PR TITLE
docs: document usage of downloader plugins

### LINKED ISSUE #266: docs: provide examples of Helm (getter) plugins
Support for Helm (getter) plugins has been worked on in #263 and will be merged soon. The mentioned PR does however not offer extensive documentation or good examples about how one should enable the plugins for their Helm operator instance.

The following information should be documented at least:

- clear example of how one can utilize an init container and a mounted `$HELM_HOME` to install _any_ getter plugin using `helm<2|3> plugin install`
- examples for the (to my knowledge) most popular getter plugins:
  - https://github.com/hypnoglow/helm-s3
  - https://github.com/hayorov/helm-gcs

### PR DESCRIPTION
**Pull request by hiddeco**
Closes: #266

### CODE REVIEW

File: docs/references/helmrelease-custom-resource.md [hunk 1]
```diff
@@ -111,6 +111,108 @@ OK
...
+downloaded plugin becomes available.
+
+```yaml
+spec:
+  containers:
+  - name: flux-helm-operator
+    image: docker.io/fluxcd/helm-operator:<tag>
+    ...
+    volumeMounts:
+    - name: helm-plugin-cache
```
Comment (stefansedich) [inline docs/references/helmrelease-custom-resource.md]: Typo here s/helm-plugin-cache/helm-plugins-cache

File: docs/references/helmrelease-custom-resource.md [hunk 2]
```diff
@@ -111,6 +111,108 @@ OK
...
+   
+```yaml
+spec:
+  initContainers:
+  - name: helm-3-downloader-plugin
+    image: docker.io/fluxcd/helm-operator:<tag>
+    imagePullPolicy: IfNotPresent
+    command: ['sh', '-c', 'helm3 plugin install <plugin> --version <version>']
+    volumeMounts:
+    - name: helm-plugin-cache
```
Comment (stefansedich) [inline docs/references/helmrelease-custom-resource.md]: Typo here s/helm-plugin-cache/helm-plugins-cache

File: docs/references/helmrelease-custom-resource.md [hunk 3]
```diff
@@ -111,6 +111,108 @@ OK
...
+The available `helm2` and `helm3` binaries can then be used to install
+the plugin.
+   
+```yaml
+spec:
+  initContainers:
+  - name: helm-3-downloader-plugin
+    image: docker.io/fluxcd/helm-operator:<tag>
+    imagePullPolicy: IfNotPresent
+    command: ['sh', '-c', 'helm3 plugin install <plugin> --version <version>']
```
Comment (stefansedich) [inline docs/references/helmrelease-custom-resource.md]: The placeholders like \<plugin\>, \<version\> etc.. were not immediately obvious to me, wonder if some bold here might help?

File: docs/references/helmrelease-custom-resource.md [hunk 4]
```diff
@@ -111,6 +111,108 @@ OK
 > either need to port forward before making the request or put something
 > in front of it to serve as a gatekeeper.
 
+## Extending the supported Helm repository protocols
+
+By default, the Helm operator is able to pull charts from repositories
+using HTTP/S. It is however possible to extend the supported protocols
+by making use of a [Helm downloader plugin](https://helm.sh/docs/topics/plugins/#downloader-plugins),
+this allows you for example to use charts hosted on [Amazon S3](https://github.com/hayorov/helm-gcs)
```
Comment (stefansedich) [inline docs/references/helmrelease-custom-resource.md]: The Amazon S3 link points to helm-gcs too.

**Review by stefansedich — CHANGES_REQUESTED** [review]
Looking good @hiddeco followed this end to end and everything worked, there were just a few typos that broke the copy+paste experience and the \<x\> placeholders were not immediately obvious to me.

**Review by stefanprodan — APPROVED** [review]
LGTM

**Pull request merged**